processing/src/main/java/org/apache/druid/query/aggregation/first/DoubleFirstAggregatorFactory.java

```diff
@@ -100,27 +106,37 @@ public DoubleFirstAggregatorFactory(
   @Override
   public Aggregator factorize(ColumnSelectorFactory metricFactory)
   {
-    final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(fieldName);
+    final ColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(fieldName);
     if (valueSelector instanceof NilColumnValueSelector) {
       return NIL_AGGREGATOR;
     } else {
       return new DoubleFirstAggregator(
           metricFactory.makeColumnValueSelector(timeColumn),
-          valueSelector
+          valueSelector,
+          StringFirstLastUtils.selectorNeedsFoldCheck(
+              valueSelector,
+              metricFactory.getColumnCapabilities(fieldName),
+              SerializablePairLongDouble.class
+          )
       );
     }
   }
 
   @Override
   public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
   {
-    final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(fieldName);
+    final ColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(fieldName);
     if (valueSelector instanceof NilColumnValueSelector) {
       return NIL_BUFFER_AGGREGATOR;
     } else {
       return new DoubleFirstBufferAggregator(
           metricFactory.makeColumnValueSelector(timeColumn),
-          valueSelector
+          valueSelector,
+          StringFirstLastUtils.selectorNeedsFoldCheck(
```

Comment:
This this is dealing with numeric capabilities probably we should move away from StringFirstLastUtils and create a separate utils for numeric. Also the fold check can be avoided

Reply:
Ack. Will create a new numeric util